Aagam_Shah/postgreAssignment/src/controller/school.controller.ts

```diff
@@ -0,0 +1,359 @@
+    return;
+  }
+};
+
+exports.studentOfTeacher = async (ctx: Context) => {
+  var id = ctx.params.id;
+  if(isValidId(id)){
+    try{
+      const response = await db.query(
+        `select * 
+        from school.student as st, school.subject as sb
+        where st.classid = sb.classid and sb.teacherid =  $1`, [id]
+      );
+      if(response.rows.length){
+        ctx.response.status = 200;
+        ctx.body = response.rows;
+        return;
+      }
+      ctx.response.status = 400;
+      ctx.body = "Required Id is not Present";
+      return;
+    }
+    catch(error){
+      ctx.response.status = 500;
+      ctx.body = error.message;
+      return;
+    }
+  }
+  else{
+    ctx.body = "invalid id passed";
+    ctx.response.status = 400;
+    return;
+  }
+};
+
+exports.studentOfSubject = async (ctx: Context) => {
+  var id = ctx.params.id;
+  if(isValidId(id)){
+    try{
+      const response = await db.query(
+        `select * 
+        from school.student as st, school.subject as sb
+        where st.classid = sb.classid and sb.subjectid = $1`, [id]
+      );
+      if(response.rows.length){
+        ctx.response.status = 200;
+        ctx.body = response.rows;
+        return;
+      }
+      ctx.response.status = 400;
+      ctx.body = "Required Id is not Present";
+      return;
+    }
+    catch(error){
+      ctx.response.status = 500;
+      ctx.body = error.message;
+      return;
+    }
+  }
+  else{
+    ctx.body = "invalid id passed";
+    ctx.response.status = 400;
+    return;
+  }
+};
+
+exports.marksOfStudent = async (ctx: Context) => {
+  var id = ctx.params.id;
+  if(isValidId(id)){
+    try{
+      const response = await db.query(
+        `select * from school.marks where studentid = $1`, [id]
+      );
+      if(response.rows.length){
+        ctx.response.status = 200;
+        ctx.body = response.rows;
+        return;
+      }
+      ctx.response.status = 400;
+      ctx.body = "Required Id is not Present";
+      return;
+    }
+    catch(error){
+      ctx.response.status = 500;
+      ctx.body = error.message;
+      return;
+    }
+  }
+  else{
+    ctx.body = "invalid id passed";
+    ctx.response.status = 400;
+    return;
+  }
+};
+
+exports.highestMarkOfSubject = async (ctx: Context) => {
+  var id = ctx.params.id;
+  if(isValidId(id)){
+    try{
+      const response = await db.query(
+        `select * from school.marks 
+        where subjectid = $1
+        order by marks desc
+        limit 1`, [id]
+      );
+      if(response.rows.length){
+        ctx.response.status = 200;
+        ctx.body = response.rows;
+        return;
+      }
+      ctx.response.status = 400;
+      ctx.body = "Required Id is not Present";
+      return;
+    }
+    catch(error){
+      ctx.response.status = 500;
+      ctx.body = error.message;
+      return;
+    }
+  }
+  else{
+    ctx.body = "invalid id passed";
+    ctx.response.status = 400;
+    return;
+  }
+};
+
+exports.topStudents = async (ctx: Context) => {
```

Comment:
we need to spilt this file into multiple controller, this file is doing lot of things. Check other pr once on the controllers created.  Check Druval pr https://github.com/adityajawade93/freshers-june2021/pull/29

Reply:
done